tests/unit/test_session.py

```diff
@@ -174,6 +175,43 @@ def test_create_w_database_role(self):
             "CloudSpanner.CreateSession", attributes=TestSession.BASE_ATTRIBUTES
         )
 
+    def test_create_session_span_annotations(self):
```

Comment:
Can we add unit tests to verify spans for different pool class similar to what is done for create_session?

Reply:
Please take a look at the additions to tests/unit/test_pool.py @harshachinta